From 398931f5ce88b633af857cb82566a4feee316d59 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:01 -0400
Subject: [usb] mos7840: fix tiocmget error handling

Message-id: <1410887463-20674-50-git-send-email-dzickus@redhat.com>
Patchwork-id: 95143
O-Subject: [RHEL7 PATCH 049/591] USB: mos7840: fix tiocmget error handling
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a91ccd26e75235d86248d018fe3779732bcafd8d
Author: Johan Hovold <jhovold@gmail.com>
Date:   Wed Oct 9 17:01:09 2013 +0200

    USB: mos7840: fix tiocmget error handling

    Make sure to return errors from tiocmget rather than rely on
    uninitialised stack data.

    Cc: stable@vger.kernel.org
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/mos7840.c b/drivers/usb/serial/mos7840.c
index 973865d..a69da83 100644
--- a/drivers/usb/serial/mos7840.c
+++ b/drivers/usb/serial/mos7840.c
@@ -1532,7 +1532,11 @@ static int mos7840_tiocmget(struct tty_struct *tty)
   return -ENODEV;
 
  status = mos7840_get_uart_reg(port, MODEM_STATUS_REGISTER, &msr);
+ if (status != 1)
+  return -EIO;
  status = mos7840_get_uart_reg(port, MODEM_CONTROL_REGISTER, &mcr);
+ if (status != 1)
+  return -EIO;
  result = ((mcr & MCR_DTR) ? TIOCM_DTR : 0)
      | ((mcr & MCR_RTS) ? TIOCM_RTS : 0)
      | ((mcr & MCR_LOOPBACK) ? TIOCM_LOOP : 0)
-- 
1.7.1